CodePush.m

```diff
@@ -468,6 +529,55 @@ - (void)savePendingUpdate:(NSString *)packageHash
 }
 
 /*
+ * This method is checks if a new status update exists (new version was installed, 
+ * or an update failed) and return its details (version label, status).
+ */
+RCT_EXPORT_METHOD(getNewStatusReport:(RCTPromiseResolveBlock)resolve
+                            rejecter:(RCTPromiseRejectBlock)reject)
+{
+    // Check if the current appVersion has been reported.
+    NSString *appVersion = [[CodePushConfig current] appVersion];
+    if ([self isDeploymentStatusNotYetReported:appVersion]) {
+        [self recordDeploymentStatusReported:appVersion
+                                      status:DeploymentSucceeded];
+        resolve(@{ @"appVersion": appVersion });
+        return;
+    }
+    
+    // Check if there was a rollback that was not yet reported
+    NSUserDefaults *preferences = [NSUserDefaults standardUserDefaults];
+    NSMutableArray *failedUpdates = [preferences objectForKey:FailedUpdatesKey];
```

Comment:
Could we just set a flag before doing a rollback that we then check here? I'd love to keep this function as simple as possible.